### ISSUE TITLE
[Feature] Ability to bootstrap from another `MariaDB` 

### ISSUE THREAD

**Issue by mmontes11** [mariadb-operator/mariadb-operator#447]
**Describe the solution you'd like**
Ability to bootstrap from another `MariaDB` by specifying a reference to another `MariaDB` instance:

```yaml
apiVersion: k8s.mariadb.com/v1alpha1
kind: MariaDB
metadata:
  name: mariadb-from-mariadb
spec:
  ...
  bootstrapFrom:
    mariaDbRef:
      name: another-mariadb
    targetRecoveryTime: 2023-12-19T09:00:00Z
```

The new `MariaDB` will have the same data as the referred `MariaDB`. Also, by default, the new `MariaDB` will inherit the configuration unless it explicitly specifies `spec.myCnf` or `spec.myCnfConfigMapKeyRef`, something which will override the referred configuration:

```yaml
apiVersion: k8s.mariadb.com/v1alpha1
kind: MariaDB
metadata:
  name: mariadb-from-mariadb
spec:
  ...
  bootstrapFrom:
    mariaDbRef:
      name: another-mariadb
    targetRecoveryTime: 2023-12-19T09:00:00Z
    
  myCnf: |
    [mariadb]
    bind-address=*
    default_storage_engine=InnoDB
    binlog_format=row
    innodb_autoinc_lock_mode=2
    max_allowed_packet=256M
```

**Additional context**
Steps to implement this:
- Create a `ConfigMap` based on the referred `MariaDB` config
- Set `spec.myCnfConfigMapKeyRef` in the new `MariaDB` pointing to the previous `ConfigMap`
-  Take a `Backup` of the referred `MariaDB` at the end of the new `MariaDB` provisioning. By default, a `Backup` is created using the referred `MariaDB` PVC template as  `mariaDbBackupStorage`, but an user will be able to define a different storage i.e. S3:
```yaml
apiVersion: k8s.mariadb.com/v1alpha1
kind: MariaDB
metadata:
  name: mariadb-from-mariadb
spec:
  ...
  bootstrapFrom:
    mariaDbRef:
      name: another-mariadb
    backupJob:
      metadata:
        labels:
          sidecar.istio.io/inject: "false"
      affinity:
          enableAntiAffinity: false
      storage:
          persistentVolumeClaim:
            resources:
                requests:
                   storage: 100Mi
                 accessModes:
                  - ReadWriteOnce
    targetRecoveryTime: 2023-12-19T09:00:00Z
```

- Restore the `Backup` as we currently do in the currently for the `bootstrapFrom` feature.

In the future, this process can potentially be simplified by supporting `VolumeSnapshots`:
- https://kubernetes.io/docs/concepts/storage/volume-snapshots/

Basically, instead of creating a `Backup` and `Restore`, we can just create the new `MariaDB` PVC from a Kubernetes-managed `VolumeSnapshot` taken from the referred `MariaDB`.

**Comment by saeedkaz** [comment]
Great, Is there a scheduled release date for this feature?

**Comment by RoodeyMental** [comment]
We would also be very interested in this feature, +1
